### ISSUE TITLE
Thellier GUI tests can no longer be called in the usual way

### ISSUE THREAD

**Issue by moonshoes87** [PmagPy/PmagPy#95]
This way of calling the Thellier GUI tests:

    if (__name__ == '__main__'):
        # set constants                                                             
        WD = os.getcwd()
        ...
        backup(project_WD)
        unittest.TextTestRunner().run(unittest.TestLoader().loadTestsFromTestCase(T\
estThellierGUI))
        revert_from_backup(project_WD)

breaks two of the common ways that are used to invoke tests.  

    python -m unittest discover # which grabs all the tests in the directory

and

    python -m unittest pmagpy_tests.test_thellier_gui

In both cases, `__name__` is not `__main__`, so the necessary constants are not set.  @Caoimhinmg , it looks like these are your changes.  I'm not exactly sure how you're running these tests, but all tests need to work in the two cases above.  There is now some documentation about running tests in pmagpy_tests/README.md

Perhaps the revert directory could be added to the tearDown function instead, and the constants could be set at the top of the file as before?  

I'm having some other issues running these updated tests even if I set constants, but one thing at a time.

**Comment by Caoimhinmg** [comment]
Oops sorry @moonshoes87 was trying to make tests run independent of data set so I could test on a bunch more data without having to worry about overwriting any previous analysis work. I'll take the if __name__=='__main__' out so it just runs tests and checks sys.argv every time and see if that fixes the issue.

**Comment by moonshoes87** [comment]
Great, thanks for fixing this.  One last quick issue -- I had to fix Thellier GUI to correctly handle the situation where no thellier_gui_preferences.py file exists in the working directory.  The tests now run as expected with no thellier_gui_preferences (which doesn't exist in Github in data_files/testing/my_project).  The only question is, should thellier_gui_preferences.py be in version control here?  Should the thellier GUI tests require preferences, or is it irrelevant in this case?

**Comment by moonshoes87** [comment]
I spoke too soon about tests working (I forgot to comment out @unittest.skip, oops).  When I actually ran the tests, I got 4 out of 7 errors.

I had to fix some incorrect menu names, and now the tests run.  I.e., changed this:

    analysis_menu = self.get_menu_from_frame(self.frame,"Analysis")

to this:

    analysis_menu = self.get_menu_from_frame(self.frame,"&Analysis")

If the thellier_gui_preferences are not required, we can close this issue for good.

**Comment by Caoimhinmg** [comment]
I have no idea how the preferences requirement came about, but I made a preferences file a while back simply to test that functionality and I completely forgot about it so I never caught the error. Thanks for that fix, I think that we should probably not require a preference file. As to the menus not getting found it looks like on my system (older wx) the menus get returned with just `name` whereas on yours they get returned as `"&"+name` so I just altered the find function to search for both so the tests always get the menu (I hope). Let me know if you've got any more issues with the tests.
